lib/backend.js

```diff
@@ -183,15 +183,19 @@ Backend.prototype.addProjection = function(name, collection, fields) {
 /**
  * Add middleware to an action or array of actions
  */
-Backend.prototype.use = function(action, fn) {
+Backend.prototype.use = function(action, fn, order) {
   if (Array.isArray(action)) {
     for (var i = 0; i < action.length; i++) {
-      this.use(action[i], fn);
+      this.use(action[i], fn, order);
     }
     return this;
   }
+  fn.__order = order || 0;
```

Comment:
I don't know how hacky/icky it is to define a property on the middleware function itself.

Ideally I would have wrapped this into an object or something:

```js
{fn, order}
```

...but `this.middleware` isn't underscored, so consumers may technically assume it as part of the public API.

If I had access to modern ES features, I'd maybe use a `WeakMap` to avoid mutating the function, but we still technically target ES3.

Reply:
There's also a bug here where if someone registers the same middleware on 2 different servers (in the same thread) with different orders, it'll overwrite the first order with the second order